plugins/kotlin-as-java/src/test/kotlin/CompanionAsJavaTest.kt

```diff
@@ -0,0 +1,471 @@
+                val parentClass = module.packages.flatMap { it.classlikes }
+                    .firstOrNull { it.name == "MyClass" } as DClass
+
+                assertEquals(JavaVisibility.Public,
+                    parentClass.properties.firstOrNull()?.visibility?.values?.first())
+                assertNull(parentClass.functions.firstOrNull { it.name.contains("lateInitProp", ignoreCase = true) },
+                    "There is no getter for the cont field")
+            }
+        }
+    }
+
+    @Test
+    fun `named companion instance property should be rendered if companion rendered`() {
+        testInline(
+            """
+            |/src/main/kotlin/kotlinAsJavaPlugin/sample.kt
+            |package kotlinAsJavaPlugin
+            |class MyClass {
+            |    companion object $COMPANION_NAME {
+            |       var property: String = ""
+            |    }
+            |}
+        """.trimMargin(),
+            configuration,
+        ) {
+            documentablesTransformationStage = { module ->
+                val parentClass = module.packages.flatMap { it.classlikes }
+                    .firstOrNull { it.name == "MyClass" } as DClass
+
+                assertNotNull(parentClass.properties.any {it.name == COMPANION_NAME})
+            }
+        }
+    }
+
+    @Test
+    fun `default named companion instance property should be rendered if companion rendered`() {
+        testInline(
+            """
+            |/src/main/kotlin/kotlinAsJavaPlugin/sample.kt
+            |package kotlinAsJavaPlugin
+            |class MyClass {
+            |    companion object {
+            |       var property: String = ""
+            |    }
+            |}
+        """.trimMargin(),
+            configuration,
+        ) {
+            documentablesTransformationStage = { module ->
+                val parentClass = module.packages.flatMap { it.classlikes }
+                    .firstOrNull { it.name == "MyClass" } as DClass
+
+                assertNotNull(parentClass.properties.any {it.name == "Companion"})
+            }
+        }
+    }
+
+    @Test
+    fun `companion instance property should be hidden if companion not rendered`() {
+        testInline(
+            """
+            |/src/main/kotlin/kotlinAsJavaPlugin/sample.kt
+            |package kotlinAsJavaPlugin
+            |class MyClass {
+            |    companion object $COMPANION_NAME {
+            |       const val property: String = ""
+            |    }
+            |}
+        """.trimMargin(),
+            configuration,
+        ) {
+            documentablesTransformationStage = { module ->
+                val parentClass = module.packages.flatMap { it.classlikes }
+                    .firstOrNull { it.name == "MyClass" } as DClass
+
+                assertNotNull(parentClass.properties.none {it.name == COMPANION_NAME})
+            }
+        }
+    }
+
+
+
+    private fun `companion object not rendered for declaration`(declaration: String) {
```

Comment:
[same comment](https://github.com/Kotlin/dokka/pull/2691#discussion_r993795261) here about callable method naming

I'd rename it to something like `assertCompanionObjectNotRendered`.

upd: oh, I see that you already have `assertCompanionNotRendered` method.. [Same comment](https://github.com/Kotlin/dokka/pull/2691#discussion_r993802472) as in the other PR as well :) 

This could be extracted into a separate helper function and re-used
```
val parentClass = module.packages.flatMap { it.classlikes }
                    .firstOrNull { it.name == "MyClass" } as DClass
```

other than that I don't see any real and harmful copy-paste

I personally would rather have tests with plain structure and _some_ copy-paste, but without having to hop around too much to understand what it checks. In current form it would also be difficult to modify these tests if 1 of 5 cases changes or requires some additional checks, and would probably lead to more input params/flags to this function

Reply:
Thanks! I've returned the plain tests. I will stick to the idea that code duplication in the tests sometimes is a better decision.